src/internal/SignalGenerator.cpp

```diff
@@ -0,0 +1,52 @@
+/**
+* @file SignalGenerator.cpp
+* @brief This header file contains the implementation functions of the SignalGenerators class.
+*
+* @author Murilo Mendonça Venâncio
+* 
+* @date 24/01/2019
+*/
+
+#include "SignalGenerator.hpp"
+
+
+SignalGenerator::SignalGenerator()
+{
+  /** These are the default values for starting the class */
+  setSignalGenerator(1,1,true);
+  m_deg  = 0.;
+}
+
+SignalGenerator::~SignalGenerator()
+{
+
+}
+
+SignalGenerator::SignalGenerator(const double &amp,const double &freq,const bool &cosine)
```

Comment:
For native variables (`double`, `int`, `float`, `char`, etc) I would not pass as reference like `const double&`. For vectors and complex data types I would do like you did.